### PR TITLE
Add schain to address map

### PR DESCRIPTION
**Pull request by payvint**

### CODE REVIEW

File: contracts/SchainsInternal.sol [hunk 1]
```diff
@@ -913,6 +913,14 @@ contract SchainsInternal is Permissions, ISchainsInternal {
         return bytes(schains[schainHash].name).length != 0;
     }
 
+    function _addAddressToSchain(bytes32 schainHash, address nodeAddress) internal virtual returns (bool) {
```
Comment (DimaStebaev) [inline contracts/SchainsInternal.sol:932]: What about adding of `schainExists` modifier?

File: contracts/SchainsInternal.sol [hunk 2]
```diff
@@ -913,6 +913,14 @@ contract SchainsInternal is Permissions, ISchainsInternal {
...
+        return _nodeAddressInSchain[schainHash].add(nodeAddress);
+    }
+
+    function _removeAddressFromSchain(bytes32 schainHash, address nodeAddress) internal virtual returns (bool) {
```
Comment (DimaStebaev) [inline contracts/SchainsInternal.sol:936]: What about adding of `schainExists` modifier?

File: contracts/test/SchainsInternalMock.sol [hunk 3]
```diff
@@ -44,4 +44,12 @@ contract SchainsInternalMock is SchainsInternal, ISchainsInternalMock {
         mapping(bytes32 => uint[]) storage schainToException = _getSchainToExceptionNodes();
         delete schainToException[schainHash];
     }
+
+    function _addAddressToSchain(bytes32, address) internal override returns (bool) {
```
Comment (DimaStebaev) [inline contracts/test/SchainsInternalMock.sol]: Must be `pure`

File: contracts/test/SchainsInternalMock.sol [hunk 4]
```diff
@@ -44,4 +44,12 @@ contract SchainsInternalMock is SchainsInternal, ISchainsInternalMock {
...
+        return true;
+    }
+
+    function _removeAddressFromSchain(bytes32, address) internal override returns (bool) {
```
Comment (DimaStebaev) [inline contracts/test/SchainsInternalMock.sol]: Must be `pure`.

File: dictionaries/names.txt [hunk 5]
```diff
@@ -13,4 +13,4 @@ payvin
 payvint
 stebaiev
 vadim
-yavorsky
+yavorsky
```
Comment (DimaStebaev) [inline dictionaries/names.txt:16]: What was changed?

File: slither.config.json [hunk 6]
```diff
@@ -1,4 +1,4 @@
 {
-    "detectors_to_exclude": "similar-names,reentrancy-events,solc-version,assembly,timestamp,calls-loop,reentrancy-no-eth,public-mappings-nested,incorrect-equality",
+    "detectors_to_exclude": "similar-names,reentrancy-events,solc-version,assembly,timestamp,calls-loop,reentrancy-no-eth,public-mappings-nested,incorrect-equality,dead-code",
```
Comment (DimaStebaev) [inline slither.config.json:2]: I'm not sure to that ignoring `dead-code` is a good idea.

File: contracts/SchainsInternal.sol [hunk 7]
```diff
@@ -82,6 +83,8 @@ contract SchainsInternal is Permissions, ISchainsInternal {
 
     uint public currentGeneration;
 
+    mapping (bytes32 => EnumerableSetUpgradeable.AddressSet) private _nodeAddressInSchain;
```
Comment (DimaStebaev) [inline contracts/SchainsInternal.sol:90]: This mapping requires initialization during an upgrade.

**Review by DimaStebaev — CHANGES_REQUESTED** [review]

**Review by DimaStebaev — COMMENTED** [review]
Looks norm.

**Review by DimaStebaev — APPROVED** [review]

**Review by yavrsky — APPROVED** [review]

**Review by kladkogex — APPROVED** [review]

**Review by gannakulikova — APPROVED** [review]

**Review by PolinaKiporenko — APPROVED** [review]

**Pull request merged**